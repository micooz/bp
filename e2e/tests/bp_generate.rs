use bp_cli::{
    commands::generate,
    options::generate::{ConfigType, GenerateOptions},
};
use e2e::fs;

#[tokio::test(flavor = "multi_thread")]
async fn test_generate_configs() {
    let config_path_client = "tests/tmp/client.config.json";
    let config_path_server = "tests/tmp/server.config.json";

    generate::run(GenerateOptions {
        config: Some(config_path_server.to_string()),
        config_type: ConfigType::Server,
        ..Default::default()
    })
    .await;

    generate::run(GenerateOptions {
        config: Some(config_path_client.to_string()),
        config_type: ConfigType::Client,
        ..Default::default()
    })
    .await;

    insta::assert_snapshot!("client", fs::read_file(config_path_client).await);
    insta::assert_snapshot!("server", fs::read_file(config_path_server).await);
}

// #[tokio::test(flavor = "multi_thread")]
// async fn test_generate_certificate() {

// }
