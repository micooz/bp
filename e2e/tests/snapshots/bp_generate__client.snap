---
source: e2e/tests/bp_generate.rs
assertion_line: 26
expression: "fs::read_file(config_path_client).await"

---
{
  "bind": "127.0.0.1:1080",
  "with_basic_auth": null,
  "server_bind": "__some_where__:3000",
  "pac_bind": null,
  "key": "__some_key__",
  "encryption": "erp",
  "acl": null,
  "pin_dest_addr": null,
  "udp_over_tcp": false,
  "dns_server": "8.8.8.8:53",
  "tls": false,
  "quic": false,
  "quic_max_concurrency": null,
  "tls_cert": null
}
